src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1411,19 +1410,34 @@ def migrate_to_multisite(self, realm_name: str, zonegroup_name: str, zone_name:
                                          http_status_code=500, component='rgw')
         except SubprocessError as error:
             raise DashboardException(error, http_status_code=500, component='rgw')
-
-        if access_key and secret_key:
-            rgw_zone_modify_cmd = ['zone', 'modify', '--rgw-zone', zone_name,
-                                   '--access-key', access_key, '--secret', secret_key]
-            try:
-                exit_code, _, err = mgr.send_rgwadmin_command(rgw_zone_modify_cmd)
-                if exit_code > 0:
-                    raise DashboardException(e=err, msg='Unable to modify zone',
-                                             http_status_code=500, component='rgw')
-            except SubprocessError as error:
-                raise DashboardException(error, http_status_code=500, component='rgw')
         self.update_period()
 
+        try:
+            user_details = self.create_system_user(username, zone_name)
+            if user_details:
+                keys = user_details['keys'][0]
+                access_key = keys['access_key']
+                secret_key = keys['secret_key']
+                if access_key and secret_key:
+                    self.modify_zone(zone_name=zone_name,
+                                    zonegroup_name=zonegroup_name,
+                                    default='true', master='true',
+                                    endpoints=zone_endpoints,
+                                    access_key=keys['access_key'],
+                                    secret_key=keys['secret_key'])
+                else:
+                    raise ValueError("Access key or secret key is missing")
```

Comment:
why are we raising a ValueError? this should also raise a DashboardException so the UI can properly catch it.